README.md

```diff
@@ -1025,93 +1074,102 @@ expect(turtles).to(containElementSatisfying(^BOOL(id object) {
 ```swift
 // Swift
 
-// Passes if actual contains substring expected:
-expect(actual).to(contain(expected))
+// Passes if 'actual' contains 'substring':
+expect(actual).to(contain(substring))
 
-// Passes if actual begins with substring:
-expect(actual).to(beginWith(expected))
+// Passes if 'actual' begins with 'prefix':
+expect(actual).to(beginWith(prefix))
 
-// Passes if actual ends with substring:
-expect(actual).to(endWith(expected))
+// Passes if 'actual' ends with 'suffix':
+expect(actual).to(endWith(suffix))
 
-// Passes if actual is an empty string, "":
+// Passes if 'actual' represents the empty string, "":
 expect(actual).to(beEmpty())
 
-// Passes if actual matches the regular expression defined in expected:
+// Passes if 'actual' matches the regular expression defined in 'expected':
 expect(actual).to(match(expected))
 ```
 
 ```objc
 // Objective-C
 
-// Passes if actual contains substring expected:
+// Passes if 'actual' contains 'substring':
 expect(actual).to(contain(expected));
 
-// Passes if actual begins with substring:
-expect(actual).to(beginWith(expected));
+// Passes if 'actual' begins with 'prefix':
+expect(actual).to(beginWith(prefix));
 
-// Passes if actual ends with substring:
-expect(actual).to(endWith(expected));
+// Passes if 'actual' ends with 'suffix':
+expect(actual).to(endWith(suffix));
 
-// Passes if actual is an empty string, "":
+// Passes if 'actual' represents the empty string, "":
 expect(actual).to(beEmpty());
 
-// Passes if actual matches the regular expression defined in expected:
+// Passes if 'actual' matches the regular expression defined in 'expected':
 expect(actual).to(match(expected))
 ```
 
-## Checking if all elements of a collection pass a condition
+## Collection Elements
+
+Nimble provides a means to check that all elements of a collection pass a given expectation.
 
 ```swift
 // Swift
 
-// with a custom function:
-expect([1,2,3,4]).to(allPass({$0 < 5}))
+// Providing a custom function:
+expect([1, 2, 3, 4]).to(allPass { $0! < 5 })
 
-// with another matcher:
-expect([1,2,3,4]).to(allPass(beLessThan(5)))
+// Composing the expectation with another matcher:
+expect([1, 2, 3, 4]).to(allPass(beLessThan(5)))
 ```
 
 ```objc
 // Objective-C
 
-expect(@[@1, @2, @3,@4]).to(allPass(beLessThan(@5)));
+expect(@[@1, @2, @3, @4]).to(allPass(beLessThan(@5)));
 ```
 
-For Swift the actual value has to be a Sequence, e.g. an array, a set or a custom seqence type.
+For Swift, the actual value must be an instance of a type conforming to `Sequence`. 
+For example, an array, a set, or a custom sequence type.
 
-For Objective-C the actual value has to be a NSFastEnumeration, e.g. NSArray and NSSet, of NSObjects and only the variant which
+For Objective-C, the actual value must be an instance of a type which implements 
+the `NSFastEnumeration` protocol.
+For example, `NSArray` and `NSSet`, of `NSObject`s and only the variant which
```

Comment:
I am unsure about what the original author intended this sentence to mean. Particularly the portion of it which references "NSObject".